auth/src/main/java/com/firebase/ui/auth/viewmodel/SingleLiveEvent.java

```diff
@@ -46,6 +47,21 @@ public void onChanged(@Nullable T t) {
         });
     }
 
+    @Override
+    public void observeForever(@NonNull Observer<T> observer) {
+        throw new UnsupportedOperationException("obseveForever not supported.");
+    }
+
+    @Override
+    public void removeObserver(@NonNull Observer<T> observer) {
+        throw new UnsupportedOperationException("removeObserver not supported.");
```

Comment:
Awww... I really wish this would work, but I'm pretty sure it'll crash if you rotate the activity (I could be wrong, so it would be a good idea to try it). Here's the source code:
```java
        public final LifecycleOwner owner;
        public final Observer<T> observer;

        LifecycleBoundObserver(LifecycleOwner owner, Observer<T> observer) {
            this.owner = owner;
            this.observer = observer;
        }

        @Override
        public void onStateChanged(LifecycleOwner source, Lifecycle.Event event) {
            if (owner.getLifecycle().getCurrentState() == DESTROYED) {
                removeObserver(observer); // <-- BOOM!
                return;
            }
        }
```

Reply:
You're right :-(

Ok in the overarching spirit of "deal with it when we get there" I have just added a class comment.

Reply:
I also overrode the methods so I could mark them `@Deprecated` which doesn't really help but can't hurt.

Reply:
Haha, SGTM 👍